### PR TITLE
refact cuda ut to facilitate automation

### PR DESCRIPTION
**Pull request by n1ck-guo**

### CODE REVIEW

File: auto_round/testing_utils.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    return importlib.util.find_spec("flash-attn") is not None
+
+def is_gptqmodel_available():
+    try:
+        require_version("gptqmodel>=2.0")
+        return True
+    except ImportError:
+        return False
+
+def is_new_version():
```
Comment (wenhuach21) [inline auto_round/testing_utils.py]: refine the function name

File: auto_round/testing_utils.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+
+def require_new_version(test_case):
+    """
+    Decorator marking a test that requires auto-round>=0.5.0.
+
+    These tests are skipped when auto-round<0.5.0.
+
+    """
+    return unittest.skipUnless(is_new_version(), "test requires auto-round>=0.5.0")(test_case)
```
Comment (wenhuach21) [inline auto_round/testing_utils.py]: same issue

File: test_cuda/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,18 @@
+accelerate
+autoawq
+auto-gptq
+datasets
+einops
+gptqmodel>=2.0
+intel-extension-for-pytorch>=2.5
+intel-extension-for-transformers
```
Comment (wenhuach21) [inline test_cuda/requirements.txt]: itrex requires <=torch 2.6

File: test_cuda/test_auto_round_format.py [hunk 4]
```diff
@@ -319,35 +328,6 @@ def test_load_gptq_model_3bits(self):
         tokenizer = AutoTokenizer.from_pretrained(model_name, trust_remote_code=True)
         self.model_infer(model, tokenizer)
 
-    def test_autoround_asym(self):
-        for bits in [2, 3, 4, 8]:
-            model = AutoModelForCausalLM.from_pretrained(self.model_name, torch_dtype="auto", trust_remote_code=True)
```
Comment (wenhuach21) [inline test_cuda/test_auto_round_format.py:324]: why was this ut deleted

Comment (n1ck-guo) [reply test_cuda/test_auto_round_format.py:324]: Repeated with another ut, same name and same content

File: test/test_cuda/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+gptqmodel>=2.0
+intel-extension-for-pytorch==2.6.0
+intel-extension-for-transformers
+lm-eval>=0.4.2,<0.5
+numpy < 2.0
+optimum
+pandas
+pillow
+py-cpuinfo
+torch<2.7.0
```
Comment (wenhuach21) [inline test/test_cuda/requirements.txt]: We shouldn't restrict the PyTorch version, as it may prevent us from identifying issues introduced in the latest release. For ITREX test, we can use a specific test to ensure compatibility.

File: test/test_cuda/requirements_vlm.txt [hunk 6]
```diff
@@ -0,0 +1,21 @@
+# git+https://github.com/haotian-liu/LLaVA.git@v1.2.2
+# pip install git+https://github.com/deepseek-ai/DeepSeek-VL2.git
+accelerate
+bitsandbytes==0.44.0 
+datasets
+einops
+flash-attn==2.6.1
+intel-extension-for-pytorch==2.3.0
```
Comment (wenhuach21) [inline test/test_cuda/requirements_vlm.txt]: same

File: test/test_cuda/requirements_vlm.txt [hunk 7]
```diff
@@ -0,0 +1,21 @@
+# git+https://github.com/haotian-liu/LLaVA.git@v1.2.2
+# pip install git+https://github.com/deepseek-ai/DeepSeek-VL2.git
+accelerate
+bitsandbytes==0.44.0 
+datasets
+einops
+flash-attn==2.6.1
```
Comment (XuehaoSun) [inline test/test_cuda/requirements_vlm.txt]: flash-attn==2.5.8

File: test/test_cuda/requirements_vlm.txt (hunk 6 above)
Comment (XuehaoSun) [inline test/test_cuda/requirements_vlm.txt]: remove intel-extension-for-pytorch

File: test/test_cuda/requirements_vlm.txt [hunk 8]
```diff
@@ -0,0 +1,21 @@
+# git+https://github.com/haotian-liu/LLaVA.git@v1.2.2
+# pip install git+https://github.com/deepseek-ai/DeepSeek-VL2.git
```
Comment (XuehaoSun) [inline test/test_cuda/requirements_vlm.txt:2]: pip install -v git+https://github.com/casper-hansen/AutoAWQ.git@v0.2.0 --no-build-isolation

**Review by XuehaoSun — CHANGES_REQUESTED** [review]

File: test/test_cuda/requirements_vlm.txt (hunk 7 above)
Comment (XuehaoSun) [inline test/test_cuda/requirements_vlm.txt]: missing protobuf==3.20.1

**Review by XuehaoSun — CHANGES_REQUESTED** [review]

File: test/test_cuda/requirements.txt [hunk 9]
```diff
@@ -0,0 +1,18 @@
+accelerate
+autoawq
+auto-gptq
+datasets
+einops
+gptqmodel>=2.0
```
Comment (XuehaoSun) [inline test/test_cuda/requirements.txt]: pip install -v git+https://github.com/ModelCloud/GPTQModel.git@v2.1.0 --no-build-isolation

File: test/test_cuda/requirements.txt [hunk 10]
```diff
@@ -0,0 +1,18 @@
+accelerate
+autoawq
```
Comment (XuehaoSun) [inline test/test_cuda/requirements.txt]: pip install -v git+https://github.com/casper-hansen/AutoAWQ.git --no-build-isolation

**Review by XuehaoSun — CHANGES_REQUESTED** [review]

File: auto_round/data_type/int.py [hunk 11]
```diff
@@ -134,10 +134,13 @@ def quant_tensor_sym_dq(
 
     scale = scale.view(-1, 1)
     zp = torch.full_like(scale, maxq)  # pylint: disable=E1130
+    scale = torch.where(scale < 0, torch.clamp(scale, max=-q_scale_thresh), torch.clamp(scale, min=q_scale_thresh))
```
Comment (wenhuach21) [inline auto_round/data_type/int.py]: Why are you using full-range symmetric quantization here? As I mentioned earlier, it may not be well-suited for double quantization. Do you have any accuracy data to support this choice?

File: auto_round/data_type/int.py [hunk 12]
```diff
@@ -134,10 +134,13 @@ def quant_tensor_sym_dq(
...
     int_w = round_ste(tensor / scale + v)
     q = torch.clamp(int_w + zp, 0, 2 ** bits - 1)
     qdq_result = (scale * (q - zp)).to(tensor.dtype)
     qdq_result = revert_tensor_by_pad(qdq_result, orig_shape=orig_shape, pad_len=pad_len)
+    if qdq_result.isnan().sum() > 0:
+        breakpoint()
```
Comment (wenhuach21) [inline auto_round/data_type/int.py]: deleted later

**Review by XuehaoSun — APPROVED** [review]
